//
// Copyright 2019 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto2";

package zetasql;

message FeatureLabelDictionaryProto {
  // Encoding map representing integer codes and their corresponding
  // feature labels.
  map<int32, string> label_mapping =
      1
      ;

  // Represents a set of encoded feature labels associated with a single query.
  message LabelSet {
    repeated int32 label = 1 [
      packed = true
    ];
  }

  // Mapping of user supplied query id to set of encoded labels for that query.
  map<int32, LabelSet> encoded_labels = 2;
}
